src/lib.rs

```diff
@@ -427,14 +427,25 @@ pub struct DataFormatDescriptorHeader {
 }
 
 impl DataFormatDescriptorHeader {
-    const LENGTH: usize = 8;
+    pub const LENGTH: usize = 8;
 
     pub const BASIC: Self = Self {
         vendor_id: 0,
         descriptor_type: 0,
         version_number: 2,
     };
 
+    pub fn as_bytes(&self, descriptor_block_size: u16) -> [u8; Self::LENGTH] {
+        let mut output = [0u8; Self::LENGTH];
+
+        let first_word = self.vendor_id | (self.descriptor_type << 17);
+        output[0..4].copy_from_slice(&first_word.to_le_bytes());
+        output[4..6].copy_from_slice(&(self.version_number as u16).to_le_bytes());
```

Comment:
Any reason this isn't just stored as a u16?

Reply:
Fixed!